molr-mole-core/src/main/java/io/molr/mole/core/support/OngoingMissionRun.java

```diff
@@ -41,4 +67,181 @@ public OngoingMissionRun and() {
                 .map(MissionState::result);
     }
 
+    /**
+     * @return a {@link Mono} of {@link MissionHandle}
+     */
+    public Mono<MissionHandle> asyncHandle() {
+        return handle;
+    }
+
+    /**
+     * @return the {@link MissionHandle} represented by this {@link OngoingMissionRun}
+     */
+    public MissionHandle awaitHandle() {
+        return handle.block();
+    }
+
+    /**
+     * @param command the {@link StrandCommand} to be given to running {@link Mission}
+     */
+    public void instruct(StrandCommand command) {
+        handle.subscribe(h -> mole.instructRoot(h, command));
+    }
+
+    /**
+     * @param runStateValidator a {@link MissionPredicates} for {@link RunState}
+     */
+    public void await(MissionPredicates<RunState> runStateValidator) {
+        filterState(runStateValidator, handle, mole)
+                .blockFirst();
+    }
+
+    /**
+     * @param runStateValidator a {@link MissionPredicates} for {@link RunState}
+     * @param timeout           the timeout before onNext signal
+     */
+    public void await(MissionPredicates<RunState> runStateValidator, Duration timeout) {
+        filterState(runStateValidator, handle, mole)
+                .blockFirst(timeout);
+    }
+
+    /**
+     * Filters the {@link MissionState}s matching the {@link RunState} {@link MissionPredicates}
+     *
+     * @param runStateValidator a {@link MissionPredicates} for {@link RunState}
+     * @param handle            the {@link MissionHandle} of running {@link Mission}
+     * @param mole              the {@link Mole} which has missions registered
+     * @return a {@link Flux} of {@link MissionState}
+     */
+    private static Flux<MissionState> filterState(MissionPredicates<RunState> runStateValidator,
```

Comment:
I think it would be more consistent to pass in here a Predicate<MissionState> (Further comments I will put in the MissionPredicates class)